src/test/lib/concurrency/stress_test.cpp

```diff
@@ -443,6 +443,50 @@ TEST_F(StressTest, NodeQueueSchedulerMultiNumaNodeTPCHQ13) {
   }
 }
 
+TEST_F(StressTest, NodeQueueSchedulerTaskGrouping) {
+  auto node_queue_scheduler = std::make_shared<NodeQueueScheduler>();
+  Hyrise::get().set_scheduler(node_queue_scheduler);
+
+  const auto worker_count = node_queue_scheduler->workers().size();
+  if (worker_count < NodeQueueScheduler::NUM_GROUPS) {
+    // With to few workers, we would not see any impact from task grouping.
+    GTEST_SKIP();
+  }
+
+  const auto multiplier = 1'000;
+  const auto task_count = multiplier * worker_count;
+
+  for (auto run = size_t{0}; run < 10; ++run) {
+    auto previous_task_id_per_group = std::vector<size_t>(NodeQueueScheduler::NUM_GROUPS, 0);
+    auto output_counter = std::atomic<size_t>{0};
+    auto concurrently_processed_groups = std::atomic<int64_t>{0};
```

Comment:
It's literally the number of concurrently processed tasks, innit? The groups are only a helper to ensure not more parallelism is use, right? But I admit that might be a matter of taste.